GitHubContents.ps1

```diff
@@ -143,3 +153,266 @@ function Get-GitHubContent
+        Justification='One or more parameters (like NoStatus) are only referenced by helper methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    param(
+        [Parameter(
+            Mandatory,
+            Position = 1)]
+        [string] $Path,
+
+        [Parameter(
+            Mandatory,
+            Position = 2)]
+        [string] $CommitMessage,
+
+        [Parameter(
+            Mandatory,
+            Position = 3)]
+        [string] $Content,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [string] $BranchName,
+
+        [string] $CommitterName,
+
+        [string] $CommitterEmail,
+
+        [string] $AuthorName,
+
+        [string] $AuthorEmail,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    $elements = Resolve-RepositoryElements -DisableValidation
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $uriFragment = "/repos/$OwnerName/$RepositoryName/contents/$Path"
+
+    $encodedContent = [Convert]::ToBase64String([System.Text.Encoding]::Unicode.GetBytes($Content))
+
+    $hashBody = @{
+        message = $CommitMessage
+        content = $encodedContent
+    }
+
+    if ($PSBoundParameters.ContainsKey('BranchName'))
+    {
+        $hashBody['branch'] = $BranchName
+    }
+
+    if ($PSBoundParameters.ContainsKey('CommitterName') -or
+        $PSBoundParameters.ContainsKey('CommitterEmail'))
+    {
+        if (![System.String]::IsNullOrEmpty($CommitterName) -and
+            ![System.String]::IsNullOrEmpty($CommitterEmail))
+        {
+            $hashBody['committer'] = @{
+                name = $CommitterName
+                email = $CommitterEmail
+            }
+        }
+        else
+        {
+            throw 'Both CommiterName and CommitterEmail need to be specified.'
+        }
+    }
+
+    if ($PSBoundParameters.ContainsKey('AuthorName') -or
+        $PSBoundParameters.ContainsKey('AuthorEmail'))
+    {
+        if (![System.String]::IsNullOrEmpty($CommitterName) -and
+            ![System.String]::IsNullOrEmpty($CommitterEmail))
+        {
+            $hashBody['author'] = @{
+                name = $AuthorName
+                email = $AuthorEmail
+            }
+        }
+        else
+        {
+            throw 'Both AuthorName and AuthorEmail need to be specified.'
+        }
+    }
+
+    if ($PSCmdlet.ShouldProcess("$BranchName branch of $RepositoryName",
+        "Set GitHub Contents on $Path"))
```

Comment:
Generally, my preference when splitting a parameter-supplied function call across multiple lines (regardless of language) is to put _each_ parameter on its own line:

```suggestion
    if ($PSCmdlet.ShouldProcess(
        "$BranchName branch of $RepositoryName",
        "Set GitHub Contents on $Path"))
```